File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailModule.java
package com.jaynewstrom.concretesample.details;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
final class DetailModule {

    private final String detailsTitle;

    DetailModule(String detailsTitle) {
        this.detailsTitle = detailsTitle;
    }

    @Provides @Named("detailsTitle") String provideDetailsTitle() {
        return detailsTitle;
    }
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailModule.kt
package com.jaynewstrom.concretesample.details

import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
internal class DetailModule(private val detailsTitle: String) {
    @Provides @Named("detailsTitle") fun provideDetailsTitle(): String {
        return detailsTitle
    }
}
